sdk/spring/spring-cloud-azure-actuator/src/main/java/com/azure/spring/cloud/actuator/cosmos/CosmosHealthIndicator.java

```diff
@@ -45,18 +46,30 @@ public CosmosHealthIndicator(CosmosAsyncClient cosmosAsyncClient, String databas
 
     @Override
     protected void doHealthCheck(Builder builder) {
-        CosmosDatabaseResponse response = this.cosmosAsyncClient.getDatabase(database)
-            .read()
-            .block(timeout);
-
-        if (response != null) {
-            LOGGER.info("The health indicator cost {} RUs, cosmos uri: {}, dbName: {}",
-                response.getRequestCharge(), endpoint, database);
-        }
-        if (response == null) {
-            builder.down();
+        if (database == null) {
+            builder.status("The option of `spring.cloud.azure.cosmos.database` is not configured!");
         } else {
-            builder.up().withDetail("database", response.getProperties().getId());
+            try {
+                CosmosDatabaseResponse response = this.cosmosAsyncClient.getDatabase(database)
+                                                                        .read()
+                                                                        .block(timeout);
+
+                if (response != null) {
+                    LOGGER.info("The health indicator cost {} RUs, cosmos uri: {}, dbName: {}",
+                        response.getRequestCharge(), endpoint, database);
+                }
+                if (response == null) {
+                    builder.down();
+                } else {
+                    builder.up().withDetail("database", response.getProperties().getId());
+                }
+            }catch (Exception e) {
+                if (e instanceof NotFoundException) {
+                    builder.status("The option of `spring.cloud.azure.cosmos.database` is not configured correctly!");
+                } else {
+                    throw e;
```

Comment:
Which class will handle this exception?

Reply:
And in this case, `builder` will have no `status`. Is this expected behavior?

Reply:
This place is to refer: KeyVaultCertificateHealthIndicator,
```

    @Override
    protected void doHealthCheck(Health.Builder builder) {
        try {
            this.certificateAsyncClient.getCertificateWithResponse("spring-cloud-azure-not-existing-certificate")
                .block(timeout);
            builder.up();
        } catch (Exception e) {
            if (e instanceof ResourceNotFoundException) {
                builder.up();
            } else {
                throw e;
            }
        }
    }
```

I will refactor this.

Reply:
How about simplify 
```
            } catch (Exception e) {
                if (e instanceof NotFoundException) {
                    builder.up().withDetail("database", "The option of `spring.cloud.azure.cosmos.database` is not "
                        + "configured correctly!");
                } else {
                    throw e;
                }
            }
```
to 
```
            } catch (NotFoundException e) {
                builder.up().withDetail("database",
                        "The option of `spring.cloud.azure.cosmos.database` is not configured correctly!");
            }
```